appinventor/appengine/src/com/google/appinventor/server/DownloadServlet.java

```diff
@@ -94,6 +99,21 @@ public class DownloadServlet extends OdeServlet {
   private static final int USERFILE_PATH_INDEX = 4;
   private static final int SPLIT_LIMIT_USERFILE = 5;
 
+  // If any file we try to export exceeds 20MB, then use the remote storage solution if
+  //   configured.
+  // If unconfigured, still use GAE, but it may fail for large files due to response
+  //   payload limit.
+  private static final int DIRECT_DOWNLOAD_MAX_FILE_SIZE = 20_000_000;
+
+  // Only use remote downloads for specific kind of downloads (and avoid using for other
+  //   files like assets, even if "larger", although impossible).
+  private static final Set<String> REMOTE_DOWNLOAD_KINDS = Set.of(
```

Comment:
I didn't have this initially, and I was just proxying all the files through Remote Storage if the size was larger. However, the purpose here is to just allow exporting projects that exceed 32MB, so I decided to limit to project-related download kinds.